### PR TITLE
OTA with HTTP (and broken HTTPS)

### PR DESCRIPTION
**Pull request by 3rwww1**

### CODE REVIEW

**Comment by 3rwww1** [comment]
@Kameeno please rebase your branch !

```bash
git fetch # pull from remote
git rebase -i origin/master #start interactive rebasing
```
Then, in the editor, remove the line labelled `Make CoolWifi's SSID/PASS arrays dynamic`

This will remove the commit in question, using the one we already have on the master branch - and which does the same thing (but better)

You will probably experience conflicts while rebasing. It's OK, just ask me if you can't manage to solve it.

File: src/core/CoolBoard.cpp [hunk 1]
```diff
@@ -32,124 +32,128 @@
...
 void CoolBoard::loop() {
-  INFO_LOG("Connecting...");
-  if (!this->isConnected()) {
-    this->connect();
-    INFO_LOG("Sending public IP...");
-    this->sendPublicIP();
-  }
-  INFO_LOG("Updating RTC...");
-  this->rtc.update();
+  if (!doOta) {
```
Comment (3rwww1) [inline src/core/CoolBoard.cpp]: no, no, no, no.

We don't need to check both in `begin` and `loop` if we `doOTA`. If we `doOTA`, we don't need to loop. We just need to upgrade. Please move upgrading logic to be called only from `begin`.

File: src/core/CoolBoard.cpp [hunk 2]
```diff
@@ -637,3 +713,57 @@ void CoolBoard::mqttCallback(char *topic, byte *payload, unsigned int length) {
   temp[length] = '\0';
   this->update(temp);
 }
+
+void CoolBoard::otaUpdate() {
+  if (this->wifiManager.wifiMulti.run() == WL_CONNECTED && doOta) {
```
Comment (3rwww1) [inline src/core/CoolBoard.cpp]: no! DRY DRY DRY, Do not Repeat Yourself! You've already checked the `doOTA` flag before, you don't need to check it again!

**Review by Kameeno — DISMISSED** [review]

**Review by Kameeno — APPROVED** [review]

**Comment by Kameeno** [comment]
Added HTTPS OTA update via mqtt

**Pull request merged**